documentation/source/troubleshooting.rst

```diff
@@ -15,14 +15,51 @@ Increasing Verbosity
 If things fail in the VPI/VHPI/FLI area, check your simulator's documentation to see if it has options to
 increase its verbosity about what may be wrong. You can then set these options on the :command:`make` command line
 as :make:var:`COMPILE_ARGS`, :make:var:`SIM_ARGS` or :make:var:`EXTRA_ARGS` (see :doc:`building` for details).
+Moreover the scheduler activity can be enhanced by setting the variable :make:var:`COCOTB_SCHEDULER_DEBUG` to
+one.
 
 
 Attaching a Debugger
 ====================
 
+C-Code
+------
+
 In order to give yourself time to attach a debugger to the simulator process before it starts to run,
 you can set the environment variable :envvar:`COCOTB_ATTACH` to a pause time value in seconds.
 If set, cocotb will print the process ID (PID) to attach to and wait the specified time before
 actually letting the simulator run.
 
 For the GNU debugger GDB, the command is :command:`attach <process-id>`.
+
+Python
+------
+
+When executing the Makefile to run a cocotb test, a python shell interpreter is called from within the
+VPI/VHPI/FLI library. Hence it is not possible to directly attach a python debugger to the python process being
+part of the simulator that uses the aforementioned library.
+
+To successfully debug your python code use the `remote_pdb`_ python package to create a :command:`pdb` instance
+accessible via a TCP socket:
```

Comment:
It's probably worth adding a comment here saying that "Using `import pdb; pdb.set_trace()` directly is also frequently not possible, due to the way that simulators interfere with stdin."

Reply:
I haven't had this kind of experience, I don't want to write about something I am not aware of.
After all you are a maintainer of this repository and can enrich the information wherever you consider it necessary.

Reply:
I thought you said you'd tried `pdb` with no success.

Reply:
Honestly I did not really get what is going under the hood, so I don't find comfortable to describe something I do not understand fully, moreover the remote debugging it should always work as it create a socket at system level, independently of the simulator it is used.
I described only the successful recipe.
I hope you are good with that.

Reply:
If it is so important to add such information, could suggest a sentence to add to my pull request?